'warnings by module' table missing in a report

### Bug description

```python
Table 'warnings by module' missing in a report
```


### Configuration

_No response_

### Command used

```shell
echo foo > module1.py
echo bar > module1.py
pylint -ry module1.py module2.py
```


### Pylint output

```python
************* Module module1
module1.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module1.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module1.py:1:0: E0602: Undefined variable 'foo' (undefined-variable)
************* Module module2
module2.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module2.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module2.py:1:0: E0602: Undefined variable 'bar' (undefined-variable)


Report
======
2 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |2      |NC         |NC         |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



6 lines have been analyzed

Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |4      |66.67 |NC       |NC         |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |2      |33.33 |NC       |NC         |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |NC       |NC         |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |NC       |NC         |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |2      |NC       |NC         |
+-----------+-------+---------+-----------+
|refactor   |0      |NC       |NC         |
+-----------+-------+---------+-----------+
|warning    |2      |NC       |NC         |
+-----------+-------+---------+-----------+
|error      |2      |NC       |NC         |
+-----------+-------+---------+-----------+



Messages
--------

+-------------------------+------------+
|message id               |occurrences |
+=========================+============+
|undefined-variable       |2           |
+-------------------------+------------+
|pointless-statement      |2           |
+-------------------------+------------+
|missing-module-docstring |2           |
+-------------------------+------------+




-----------------------------------
Your code has been rated at 0.00/10
```


### Expected behavior
```
************* Module module1
module1.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module1.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module1.py:1:0: E0602: Undefined variable 'foo' (undefined-variable)
************* Module module2
module2.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module2.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module2.py:1:0: E0602: Undefined variable 'bar' (undefined-variable)


Report
======
2 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |2      |2          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



6 lines have been analyzed

Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |4      |66.67 |4        |=          |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |2      |33.33 |2        |=          |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |0          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |2      |2        |2          |
+-----------+-------+---------+-----------+
|refactor   |0      |0        |0          |
+-----------+-------+---------+-----------+
|warning    |2      |2        |2          |
+-----------+-------+---------+-----------+
|error      |2      |2        |2          |
+-----------+-------+---------+-----------+



% errors / warnings by module
-----------------------------

+--------+------+--------+---------+-----------+
|module  |error |warning |refactor |convention |
+========+======+========+=========+===========+
|module2 |50.00 |50.00   |0.00     |50.00      |
+--------+------+--------+---------+-----------+
|module1 |50.00 |50.00   |0.00     |50.00      |
+--------+------+--------+---------+-----------+



Messages
--------

+-------------------------+------------+
|message id               |occurrences |
+=========================+============+
|undefined-variable       |2           |
+-------------------------+------------+
|pointless-statement      |2           |
+-------------------------+------------+
|missing-module-docstring |2           |
+-------------------------+------------+




------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.8 (main, Nov 10 2022, 12:59:17) [GCC 11.2.1 20220219]
```


### OS / Environment

Alpine Linux v3.16.3 ([Dockerfile](https://github.com/cytopia/docker-pylint/blob/0.8/Dockerfiles/Dockerfile))

### Additional dependencies

_No response_

--- a/doc/whatsnew/3/3.2/index.rst
+++ b/doc/whatsnew/3/3.2/index.rst
@@ -14,6 +14,71 @@ Summary -- Release highlights
 
 .. towncrier release notes start
 
+What's new in Pylint 3.2.1?
+---------------------------
+Release date: 2024-05-18
+
+
+False Positives Fixed
+---------------------
+
+- Exclude if/else branches containing terminating functions (e.g. `sys.exit()`)
+  from `possibly-used-before-assignment` checks.
+
+  Closes #9627 (`#9627 <https://github.com/pylint-dev/pylint/issues/9627>`_)
+
+- Don't emit ``typevar-name-incorrect-variance`` warnings for PEP 695 style TypeVars.
+  The variance is inferred automatically by the type checker.
+  Adding ``_co`` or ``_contra`` suffix can help to reason about TypeVar.
+
+  Refs #9638 (`#9638 <https://github.com/pylint-dev/pylint/issues/9638>`_)
+
+- Fix a false positive for `possibly-used-before-assignment` when using
+  `typing.assert_never()` (3.11+) to indicate exhaustiveness.
+
+  Closes #9643 (`#9643 <https://github.com/pylint-dev/pylint/issues/9643>`_)
+
+
+
+Other Bug Fixes
+---------------
+
+- Fix a false negative for ``--ignore-patterns`` when the directory to be linted is specified using a dot(``.``) and all files are ignored instead of only the files whose name begin with a dot.
+
+  Closes #9273 (`#9273 <https://github.com/pylint-dev/pylint/issues/9273>`_)
+
+- Restore "errors / warnings by module" section to report output (with `-ry`).
+
+  Closes #9145 (`#9145 <https://github.com/pylint-dev/pylint/issues/9145>`_)
+
+- ``trailing-comma-tuple`` should now be correctly emitted when it was disabled globally
+  but enabled via local message control, after removal of an over-optimisation.
+
+  Refs #9608. (`#9608 <https://github.com/pylint-dev/pylint/issues/9608>`_)
+
+- Add `--prefer-stubs=yes` option to opt-in to the astroid 3.2 feature
+  that prefers `.pyi` stubs over same-named `.py` files. This has the
+  potential to reduce `no-member` errors but at the cost of more errors
+  such as `not-an-iterable` from function bodies appearing as `...`.
+
+  Defaults to `no`.
+
+  Closes #9626
+  Closes #9623 (`#9626 <https://github.com/pylint-dev/pylint/issues/9626>`_)
+
+
+
+Internal Changes
+----------------
+
+- Update astroid version to 3.2.1. This solves some reports of ``RecursionError``
+  and also makes the *prefer .pyi stubs* feature in astroid 3.2.0 *opt-in*
+  with the aforementioned ``--prefer-stubs=y`` option.
+
+  Refs #9139 (`#9139 <https://github.com/pylint-dev/pylint/issues/9139>`_)
+
+
+
 What's new in Pylint 3.2.0?
 ---------------------------
 Release date: 2024-05-14

--- a/doc/whatsnew/fragments/9139.internal
+++ b/doc/whatsnew/fragments/9139.internal
@@ -1,5 +1,0 @@
-Update astroid version to 3.2.1. This solves some reports of ``RecursionError``
-and also makes the *prefer .pyi stubs* feature in astroid 3.2.0 *opt-in*
-with the aforementioned ``--prefer-stubs=y`` option.
-
-Refs #9139

--- a/doc/whatsnew/fragments/9145.bugfix
+++ b/doc/whatsnew/fragments/9145.bugfix
@@ -1,3 +1,0 @@
-Restore "errors / warnings by module" section to report output (with `-ry`).
-
-Closes #9145

--- a/doc/whatsnew/fragments/9273.false_negative
+++ b/doc/whatsnew/fragments/9273.false_negative
@@ -1,3 +1,0 @@
-Fix a false negative for ``--ignore-patterns`` when the directory to be linted is specified using a dot(``.``) and all files are ignored instead of only the files whose name begin with a dot.
-
-Closes #9273

--- a/doc/whatsnew/fragments/9608.bugfix
+++ b/doc/whatsnew/fragments/9608.bugfix
@@ -1,4 +1,0 @@
-``trailing-comma-tuple`` should now be correctly emitted when it was disabled globally
-but enabled via local message control, after removal of an over-optimisation.
-
-Refs #9608.

--- a/doc/whatsnew/fragments/9626.bugfix
+++ b/doc/whatsnew/fragments/9626.bugfix
@@ -1,9 +1,0 @@
-Add `--prefer-stubs=yes` option to opt-in to the astroid 3.2 feature
-that prefers `.pyi` stubs over same-named `.py` files. This has the
-potential to reduce `no-member` errors but at the cost of more errors
-such as `not-an-iterable` from function bodies appearing as `...`.
-
-Defaults to `no`.
-
-Closes #9626
-Closes #9623

--- a/doc/whatsnew/fragments/9627.false_positive
+++ b/doc/whatsnew/fragments/9627.false_positive
@@ -1,4 +1,0 @@
-Exclude if/else branches containing terminating functions (e.g. `sys.exit()`)
-from `possibly-used-before-assignment` checks.
-
-Closes #9627

--- a/doc/whatsnew/fragments/9638.false_positive
+++ b/doc/whatsnew/fragments/9638.false_positive
@@ -1,5 +1,0 @@
-Don't emit ``typevar-name-incorrect-variance`` warnings for PEP 695 style TypeVars.
-The variance is inferred automatically by the type checker.
-Adding ``_co`` or ``_contra`` suffix can help to reason about TypeVar.
-
-Refs #9638

--- a/doc/whatsnew/fragments/9643.false_positive
+++ b/doc/whatsnew/fragments/9643.false_positive
@@ -1,4 +1,0 @@
-Fix a false positive for `possibly-used-before-assignment` when using
-`typing.assert_never()` (3.11+) to indicate exhaustiveness.
-
-Closes #9643

--- a/pylint/__pkginfo__.py
+++ b/pylint/__pkginfo__.py
@@ -9,7 +9,7 @@ It's updated via tbump, do not modify.
 
 from __future__ import annotations
 
-__version__ = "3.2.0"
+__version__ = "3.2.1"
 
 
 def get_numversion_from_version(v: str) -> tuple[int, int, int]:

--- a/tbump.toml
+++ b/tbump.toml
@@ -1,7 +1,7 @@
 github_url = "https://github.com/pylint-dev/pylint"
 
 [version]
-current = "3.2.0"
+current = "3.2.1"
 regex = '''
 ^(?P<major>0|[1-9]\d*)
 \.

--- a/towncrier.toml
+++ b/towncrier.toml
@@ -1,5 +1,5 @@
 [tool.towncrier]
-version = "3.2.0"
+version = "3.2.1"
 directory = "doc/whatsnew/fragments"
 filename = "doc/whatsnew/3/3.2/index.rst"
 template = "doc/whatsnew/fragments/_template.rst"

